homeassistant/components/thermostat/zwave.py

```diff
@@ -109,6 +134,24 @@ def target_temperature(self):
     def set_temperature(self, temperature):
         """Set new target temperature."""
         # set point
+        if self._zxt_120 == 1:
```

Comment:
Now we're getting into murky waters. I don't like hacking the thermostat
platform with all these extra services, it will become messy. I prefer in
that case that you create a new platform, zwave_zxt_120. That way it will
all be clear.

On Thu, Apr 7, 2016, 07:51 John Arild Berentsen <notifications@github.com>
wrote:

> In homeassistant/components/thermostat/zwave.py
> <https://github.com/balloob/home-assistant/pull/1713#discussion_r58884810>
> :
>
> > @@ -109,6 +134,24 @@ def target_temperature(self):
> >      def set_temperature(self, temperature):
> >          """Set new target temperature."""
> >          # set point
> > +        if self._zxt_120 == 1:
>
> The return is just forgotten. Regarding the separate method, this device
> support some modes/settings not available in the thermostat component. But
> it requires some new services. Is this something that could be of interest?
> I am thinking in the lines of the thermostat component have available
> methods and services for fan modes, not just on off, mode settings, swing
> control. Thoughts? 7. apr. 2016 00:59 skrev "Paulus Schoutsen" <
> notifications@github.com>:
> … <#m_-5973905868483503063_>
> In homeassistant/components/thermostat/zwave.py <#1713 (comment)
> <https://github.com/balloob/home-assistant/pull/1713#discussion_r58796610>>
> : > @@ -109,6 +134,24 @@ def target_temperature(self): > def
> set_temperature(self, temperature): > """Set new target temperature.""" > #
> set point > + if self._zxt_120 == 1: I guess it's fine to add it but let's
> extract it into a separate method. Also, you're not returning at the end of
> this if, should normal behavior still happen? — You are receiving this
> because you modified the open/close state. Reply to this email directly or
> view it on GitHub <
> https://github.com/balloob/home-assistant/pull/1713/files/60e8a60d8db7fd9ad4663ca6bdcc98c6d330272b#r58796610>
>
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly or view it on GitHub
> <https://github.com/balloob/home-assistant/pull/1713/files/60e8a60d8db7fd9ad4663ca6bdcc98c6d330272b#r58884810>
>